src/pudl/transform/eia.py

```diff
@@ -1029,12 +1019,17 @@ def core_eia860__assn_boiler_generator(context, **clean_dfs) -> pd.DataFrame:
         .drop_duplicates()
     )
 
-    bga_out = pd.merge(
-        left=bga_out,
-        right=bga_w_units,
-        how="left",
-        on=["plant_id_eia", "generator_id", "boiler_id"],
-    ).astype({"unit_id_pudl": pd.Int64Dtype()})
+    bga_out = (
+        pd.merge(
+            left=bga_out,
+            right=bga_w_units,
+            how="left",
+            on=["plant_id_eia", "generator_id", "boiler_id"],
+        )
+        .astype({"unit_id_pudl": pd.Int64Dtype()})
+        .pipe(apply_pudl_dtypes, group="eia")
+        .pipe(PUDL_PACKAGE.encode)
```

Comment:
I think the BGA table was previously getting encoded by one of those calls inside the harvesting process, and it doesn't get passed in in the same way as all the other tables, so we ended up with some FK constraint errors, and so I'm adding it in here.

Reply:
yes that makes sense. the bga asset is entirely separate from harvesting or the _core -> core assets

Reply:
remove this from the bottom and add it the the `clean_dfs` at the top of this function